File: LeetCodeProblemsGo/LeetCode1716CalculateMoneyInLeetcodeBank/LeetCode1716CalculateMoneyInLeetcodeBank.go
package LeetCode1716CalculateMoneyInLeetcodeBank

func totalMoney(n int) int {
	res := n/7*28 + (n/7-1)*(n/7)*7/2 + (n%7)*((n%7)+1)/2
	if n/7 > 0 {
		res += (n % 7) * (n / 7)
	}
	return res
}

File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode1716CalculateMoneyInLeetcodeBank.cpp
#include "LeetCode1716CalculateMoneyInLeetcodeBank.h"

#include <iostream>

int LeetCode1716CalculateMoneyInLeetcodeBank::totalMoney(int n)
{
    auto res=   n/7*28;
    res+=(n/7-1)*(n/7)*7/2;
    res+=(n%7)*((n%7)+1)/2;
    res+=+(n/7>0)?(n/7)*(n%7):0;
    return res;
}

void LeetCode1716CalculateMoneyInLeetcodeBank::Run()
{
    const int n = 10;
    std::cout<< totalMoney(n)<<'\n';
}
